tools/swift-inspect/Sources/swift-inspect/main.swift

```diff
@@ -45,31 +65,62 @@ internal struct BacktraceOptions: ParsableArguments {
 internal struct GenericMetadataOptions: ParsableArguments {
   @Flag(help: "Show allocations in mangled form")
   var mangled: Bool = false
+
+  @Flag(help: "Output JSON")
+  var json: Bool = false
+
+  #if os(iOS) || os(macOS) || os(tvOS) || os(watchOS)
+  @Flag(help: "Print out a summary from all process output")
+  #endif
+  var summary: Bool = false
+
+  @Option(help: "Output to a file")
+  var outputFile: String? = nil
 }
 
 internal func inspect(options: UniversalOptions,
                       _ body: (any RemoteProcess) throws -> Void) throws {
-  guard let processId = process(matching: options.nameOrPid) else {
-    print("No process found matching \(options.nameOrPid)")
-    return
+  if let nameOrPid = options.nameOrPid {
+    guard let processId = process(matching: nameOrPid) else {
+      print("No process found matching \(nameOrPid)", to: &Std.err)
+      return
+    }
+    guard let process = getRemoteProcess(processId: processId,
+                                         options: options) else {
+      print("Failed to create inspector for process id \(processId)", to: &Std.err)
+      return
+    }
+    try body(process)
   }
-
+  else {
 #if os(iOS) || os(macOS) || os(tvOS) || os(watchOS)
-  guard let process = DarwinRemoteProcess(processId: processId,
-                                          forkCorpse: options.forkCorpse) else {
-    print("Failed to create inspector for process id \(processId)")
-    return
-  }
-#elseif os(Windows)
-  guard let process = WindowsRemoteProcess(processId: processId) else {
-    print("Failed to create inspector for process id \(processId)")
-    return
-  }
-#else
-#error("Unsupported platform")
+    if let processIdentifiers = getAllProcesses(options: options) {
+      let totalCount = processIdentifiers.count
+      var successfulCount = 0
+      try processIdentifiers.enumerated().forEach { index, processIdentifier in
```

Comment:
I am contractually obligated to request this to be written as a for/in loop instead.

```suggestion
      for (index, processIdentifier) in processIdentifiers.enumerated() {
```